### ISSUE TITLE
Add Preferences dialog

### ISSUE THREAD

**Issue by bittner** [bittner/gedit-reST-plugin#17]
As suggested in PR #16, we should add a <kbd>Preferences</kbd> dialog to this plugin to allow user control over the behavior of the plugin.

Example Options
-------------------

| Document display |
|--|
| &nbsp; &nbsp; (o) Bottom panel |
| &nbsp; &nbsp; ( ) Side panel |

Code Examples
------------------

- [Python Console](https://gitlab.gnome.org/GNOME/gedit/-/blob/master/plugins/pythonconsole/pythonconsole/config.ui)
- [Insert Date/Time](https://gitlab.gnome.org/GNOME/gedit/-/tree/master/plugins/time/resources/ui)
- [rhythmbox-baidu-music](https://github.com/pandasunny/rhythmbox-baidu-music/commit/50089147a4090b6ab664d3242b585a31105a4049)

**Comment by JanKanis** [comment]
I also had a look myself at how GSettings/dconf works. From the documentation I got the idea that the key names are supposed to be a reversed uri of the owner of the code, so the `org.gnome.gedit.plugins.*` namespace would be reserved for plugins maintained by the GEdit team, and your key should be named `it.bittner.something`.

I also came up with a schema of
```
<?xml version="1.0" encoding="UTF-8"?>
<schemalist>
        <schema id="it.bittner.gedit-rest-plugin" path="/org/gnome/gedit/plugins/reST/">
                <key type="s" name="panelchoice">
                        <choices>
                                <choice value="side"/>
                                <choice value="bottom"/>
                        </choices>
                        <default>'bottom'</default>
                        <summary>reST preview location</summary>
                        <description>Where to show the reST preview, in the side panel or the bottom panel</description>
                </key>
        </schema>
</schemalist>
```
which makes the key more readable/editable in dconf or for someone not knowing the code itself.

**Comment by bittner** [comment]
Okay, that looks like a solution that makes much more sense.

TBH, I picked the numbers, because that seemed to be less prone to typos (numeric index vs. a string value, as I originally envisioned). I was a lazy solution, after all. If we can define the possible values in the schema that's exactly what a sound solution must look like!

Can you quote a source of where the schema is defined?

### Local schema is a probably hack

What I figured myself is that having the schema compiled and read from local is probably just a (dirty) hack. The [dconf-editor](https://wiki.gnome.org/Apps/DconfEditor) doesn't recognize this as an existing schema:

![image](https://user-images.githubusercontent.com/665072/97817040-1d79e080-1c9a-11eb-9c4d-9e06fcfc9c55.png)

It's probably better to deploy the schema properly, the standard way. TBH, this is based on assumptions. I have not consulted related documentation. Have you seen official recommendations?

**Comment by bittner** [comment]
Just for the record, PR #18 has implemented a first working draft. The screenshots in the [README](https://github.com/bittner/gedit-reST-plugin/blob/master/README.rst) show the current state.

**Comment by bittner** [comment]
I posted a [related question](https://discourse.gnome.org/t/gedit-panels-move-content-of-bottom-and-side-panel/4725) on GNOME's Discourse.

**Comment by JanKanis** [comment]
I got this schema from examples at https://developer.gnome.org/gio/stable/GSettings.html. I couldn't find a good reference document of what exactly is allowed in the schema xml except for the DTD at https://gitlab.gnome.org/GNOME/glib/-/blob/master/gio/gschema.dtd, which is not so nice to read.

I agree the local schema is a bit hackish. Unfortunately gsettings by default only looks for schemas in a few root-owned directories (the `glib-2.0/schemas/` subdirectory of all directories in `$XDG_DATA_DIRS`). Requiring root access to install an editor plugin did not look like a very user friendly idea, especially for users on shared systems who don't have root access. The question at https://askubuntu.com/questions/251712/how-can-i-install-a-gsettings-schema-without-root-privileges gives two alternatives: install in a local directory as the current code does, or set an environment variable (`GSETTINGS_SCHEMA_DIR`) to the location of the schema. Setting an environment variable is also not a great idea for a plugin, so the local schema is the only remaining option. The alternative IMO would be to just use a plain old text config file to store the setting, but then you don't have the signals integration, and it doesn't give you dconf integration either.

**Comment by JanKanis** [comment]
dconf probably recognizes the keys if you explicitly point it to the schema file.

**Comment by bittner** [comment]
Fixed via #20. :1st_place_medal: :rocket: 

Absolutely awesome work, @JanKanis!!  :trophy: :clinking_glasses:  :champagne:
